staging/src/k8s.io/client-go/tools/cache/reflector_data_consistency_detector.go

```diff
@@ -47,27 +51,27 @@ func init() {
 //
 // Note that this function will panic when data inconsistency is detected.
 // This is intentional because we want to catch it in the CI.
-func checkWatchListConsistencyIfRequested(stopCh <-chan struct{}, identity string, lastSyncedResourceVersion string, listerWatcher Lister, store Store) {
+func checkWatchListConsistencyIfRequested(ctx context.Context, identity string, lastSyncedResourceVersion string, listItemsFn listItemsFunc, watchListItemsFn watchListItemsFunc) {
```

Comment:
In the next PR I will move the method to a new pkg (client-go/utils ?) and I will export the function.

Reply:
Why do you want to move it?

Reply:
Assuming we would like to have more than one consumer of this function and knowing that the detector doesn't belong to the reflector only to the watchlist functionality having it in different place is not incorrect.